sdv/cag/range.py

```diff
@@ -0,0 +1,376 @@
+        )
+
+        return data
+
+    def _reverse_transform(self, data):
+        """Reverse transform the table data.
+
+        The reverse transformation consists in replacing `low_diff_column` and
+        `high_diff_column` by the sum of `low` and `low_diff_column` and `middle`
+        and `high_diff_column` respectively.
+
+        Args:
+            data (dict[str, pd.DataFrame]):
+                Table data.
+
+        Returns:
+            dict[str, pd.DataFrame]:
+                Transformed data.
+        """
+        table_name = self._get_single_table_name(self.metadata)
+        table_data = data[table_name]
+        low_diff_column = np.exp(table_data[self._low_diff_column_name]) - 1
+        high_diff_column = np.exp(table_data[self._high_diff_column_name]) - 1
+        if self._dtype != np.dtype('float'):
+            low_diff_column = low_diff_column.round()
+            high_diff_column = high_diff_column.round()
+
+        if self._is_datetime:
+            low_diff_column = _convert_to_timedelta(low_diff_column)
+            high_diff_column = _convert_to_timedelta(high_diff_column)
+
+        low = table_data[self._low_column_name].to_numpy()
+        if self._is_datetime and self._dtype == 'O':
+            low = cast_to_datetime64(low, self._low_datetime_format)
+
+        middle = pd.Series(low_diff_column + low).astype(self._dtype)
+        table_data[self._middle_column_name] = middle
+        table_data[self._high_column_name] = pd.Series(high_diff_column + middle.to_numpy()).astype(
+            self._dtype
+        )
+
+        if self._nan_column_name in table_data.columns:
+            table_data = revert_nans_columns(table_data, self._nan_column_name)
+
+        data[table_name] = table_data.drop(
+            [self._low_diff_column_name, self._high_diff_column_name], axis=1
+        )
+
+        return data
+
+    def _is_valid(self, data):
+        """Check whether the `middle` column is between the `low` and `high` columns.
+
+        Args:
+            data (dict[str, pd.DataFrame]):
+                Table data.
+
+        Returns:
+            dict[str, pd.Series]:
+                Whether each row is valid.
+        """
+        table_name = self._get_single_table_name(self.metadata)
+        is_valid = {
+            table: pd.Series(True, index=table_data.index)
+            for table, table_data in data.items()
+            if table != table_name
+        }
+
+        table_data = data[table_name]
+        low, mid, high = self._get_data(table_data)
+        low_is_nan = pd.isna(low)
+        mid_is_nan = pd.isna(mid)
+        high_is_nan = pd.isna(high)
+
+        low_lt_middle = self._operator(low, mid) | low_is_nan | mid_is_nan
+        mid_lt_high = self._operator(mid, high) | mid_is_nan | high_is_nan
+        low_lt_high = self._operator(low, high) | low_is_nan | high_is_nan
+
+        valid = low_lt_middle & mid_lt_high & low_lt_high
+        is_valid[table_name] = valid
```

Comment:
Given the common code here and in `_validate_pattern_with_data`, can we create a function?